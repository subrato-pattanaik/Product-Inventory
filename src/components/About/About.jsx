import React from "react";
import { Carousel, Container } from "react-bootstrap";
import EditProduct from "../../images/EditProduct.jpg";
import NewProduct from "../../images/NewProduct.jpg";
import PI from "../../images/PI.jpg";
import Viewproduct from "../../images/Viewproduct.jpg";
function About() {
  return (
    <Container fluid className="p-0" style={{ Height: "100vh !important" }}>
      <Carousel>
        <Carousel.Item>
          <img className="d-block w-100" src={PI} alt="First slide" />
        </Carousel.Item>
        <Carousel.Item>
          <img className="d-block w-100" src={NewProduct} alt="Second slide" />
        </Carousel.Item>
        <Carousel.Item>
          <img className="d-block w-100" src={EditProduct} alt="Third slide" />
        </Carousel.Item>

        <Carousel.Item>
          <img
            className="d-block w-100"
            src={Viewproduct}
            alt="TFourth slide"
          />
        </Carousel.Item>
      </Carousel>
    </Container>
  );
}

export default About;
