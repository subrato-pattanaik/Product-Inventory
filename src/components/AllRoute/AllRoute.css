.loader {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  border: 4px solid #f3f3f3;
  box-sizing: border-box;
  border-radius: 50%;
  border-top: 4px solid #ffc107;
  border-left: 4px solid #ffc107;
  width: 60px;
  height: 60px;
  animation: spin 1s linear infinite;
  -webkit-border-radius: 50%;
  -moz-border-radius: 50%;
  -ms-border-radius: 50%;
  -o-border-radius: 50%;
  -webkit-animation: spin 1s linear infinite;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
.wrapperLoad {
  min-height: 100vh;
  background: #343a40;
}
