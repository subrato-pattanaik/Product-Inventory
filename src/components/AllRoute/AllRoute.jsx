import React, { Suspense, useState } from "react";
import { Route, Switch, withRouter } from "react-router-dom";
import "./AllRoute.css";
import { UserContextAPI } from "../ContextAPI/UserContextAPI";

const Login = React.lazy(() => import("../Login/Login"));
const Header = React.lazy(() => import("../NavbarHeader/Header"));
const About = React.lazy(() => import("../About/About"));
const ProductList = React.lazy(() => import("../ProductList/ProductList"));
const AddProduct = React.lazy(() => import("../AddProduct/AddProduct"));
const TopViewProduct = React.lazy(() =>
  import("../TopViewProduct/TopViewProduct")
);
const Signup = React.lazy(() => import("../Signup/Signup"));
const Profile = React.lazy(() => import("../Profile/Profile"));
const EditProduct = React.lazy(() => import("../EditProduct/EditProduct"));
const ViewProduct = React.lazy(() => import("../ViewProduct/ViewProduct"));

function AllRoute() {
  const [user, setUser] = useState(null);
  return (
    <div>
      <Suspense
        fallback={
          <div className="wrapperLoad">
            <div className="loader"></div>
          </div>
        }
      >
        <UserContextAPI.Provider value={{ user, setUser }}>
          <Header />
          <Switch>
            <Route exact path="/" component={About} />
            <Route path="/topViewProduct" component={TopViewProduct} />
            <Route path="/productList" component={ProductList} />
            <Route path="/addProduct" component={AddProduct} />
            <Route path="/Login" component={Login} />
            <Route path="/Signup" component={Signup} />
            <Route path="/Profile" component={Profile} />
            <Route path="/editProduct" component={EditProduct} />
            <Route path="/viewProduct" component={ViewProduct} />
          </Switch>
        </UserContextAPI.Provider>
      </Suspense>
    </div>
  );
}

export default withRouter(AllRoute);
