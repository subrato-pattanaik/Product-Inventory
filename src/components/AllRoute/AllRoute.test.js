import { mount, shallow } from "enzyme";
import React from "react";
import { BrowserRouter, Route } from "react-router-dom";
import About from "../About/About";
import toJson from "enzyme-to-json";
import AllRoute from "./AllRoute";

let pathMap = [];
describe("all the test of AllRoute page", () => {
  it("should test snapshot of AllRoute page", () => {
    const wrapper = shallow(<AllRoute />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should renders the AllRoute component", () => {
    const wrapper = shallow(<AllRoute />);
    expect(wrapper).toBeTruthy();
  });

  it("should show Suspense component", () => {
    const component = shallow(<AllRoute />);
    expect(component.find("Suspense")).toHaveLength(1);
  });

  it("should show 9 route components", () => {
    const component = shallow(<AllRoute />);
    expect(component.find("Route")).toHaveLength(9);
  });

  let component = "";
  beforeAll(async () => {
    component = mount(
      <BrowserRouter>
        <AllRoute />
      </BrowserRouter>
    );

    pathMap = await component.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.component;
      return pathMap;
    }, {});
  });
  it("should show About component for / router ", () => {
    console.log(pathMap);
    expect(pathMap["/"]._result).toBe(About);
  });

  afterAll(() => {
    component.unmount();
  });
});
