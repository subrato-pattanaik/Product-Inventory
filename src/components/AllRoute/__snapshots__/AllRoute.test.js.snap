// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all the test of AllRoute page should test snapshot of AllRoute page 1`] = `
<div>
  <Suspense
    fallback={
      <div
        className="wrapperLoad"
      >
        <div
          className="loader"
        />
      </div>
    }
  >
    <ContextProvider
      value={
        Object {
          "setUser": [Function],
          "user": null,
        }
      }
    >
      <lazy />
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": [Function],
              "_status": 1,
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/topViewProduct"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/productList"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/addProduct"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/Login"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/Signup"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/Profile"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/editProduct"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/viewProduct"
        />
      </Switch>
    </ContextProvider>
  </Suspense>
</div>
`;
